packages/siop-oid4vp/lib/types/Events.ts

```diff
@@ -26,21 +26,27 @@ export enum AuthorizationEvents {
 
 export class AuthorizationEvent<T> {
   private readonly _subject: T | undefined
+  private readonly _redirectUri: string | undefined
   private readonly _error?: Error
   private readonly _timestamp: number
   private readonly _correlationId: string
 
-  public constructor(args: { correlationId: string; subject?: T; error?: Error }) {
+  public constructor(args: { correlationId: string; subject?: T; redirectUri?: string, error?: Error }) {
     //fixme: Create correlationId if not provided. Might need to be deferred to registry though
     this._correlationId = args.correlationId
     this._timestamp = Date.now()
     this._subject = args.subject
+    this._redirectUri = args.redirectUri
```

Comment:
Huh, we have a generic subject. Why are we creating something very specific (a redirectUri) on every event type. That should not happen

Reply:
It's optional and it's not every event type, just AuthorizationEvent types.
```
await this.emitEvent(AuthorizationEvents.ON_AUTH_RESPONSE_SENT_SUCCESS, { correlationId, subject: response, redirectUri: redirectUri })
```
Alternatively I can create a new type wrapping response & redirectUri. That would be a breaking change, but I do not think we are using that event anywhere else in our codebase.

Reply:
I just saw there is a simpler solution without using the event. Will revert